ios/NebulaNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -28,76 +28,96 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
     }
     
     private func start(completionHandler: @escaping (Error?) -> Void) {
-        let proto = self.protocolConfiguration as! NETunnelProviderProtocol
-        var config: Data
-        var key: String
-
-        do {
-            site = try Site(proto: proto)
-            config = try site!.getConfig()
-        } catch {
-            //TODO: need a way to notify the app
-            log.error("Failed to render config from vpn object")
-            return completionHandler(error)
-        }
+        // Load vpn configs from system, and find the manager matching the one being started
+        NETunnelProviderManager.loadAllFromPreferences() { managers, error in
+            let targetProtoConfig = self.protocolConfiguration as? NETunnelProviderProtocol;
+            let targetID = targetProtoConfig?.providerConfiguration!["id"] as? String;
```

Comment:
It would be more ideal if we didn't explode when trying to get the `id` here and in the loop below and instead called `completionHandler(with the error)`